framework/src/modules/chain/chain.js

```diff
@@ -188,6 +188,89 @@ module.exports = class Chain {
 		}
 	}
 
+	get actions() {
+		return {
+			calculateSupply: action => blockReward.calcSupply(action.params[0]),
+			calculateMilestone: action => blockReward.calcMilestone(action.params[0]),
+			calculateReward: action => blockReward.calcReward(action.params[0]),
+			generateDelegateList: action =>
+				new Promise((resolve, reject) => {
+					this.scope.modules.delegates.generateDelegateList(
+						action.params[0],
```

Comment:
maybe deconstruct like
`const [meaningfulName0, meaningfulName1] = action.params;`, same for below

Reply:
I agree with you. I created a separate issue to solve the `action.params` problem and improve readability. https://github.com/LiskHQ/lisk/issues/2991